README.md
#### Gemaakt door Cyllos

### Installatie
#### Het script is goedgekeurd door InnoGames en mag dus gebruikt worden.

Het is gemakkelijk geinstalleerd: je hebt enkel Tampermonkey nodig!

[Daarna klik je hier om te installeren.](https://github.com/Cyllos42/GME/raw/master/GrepolisMapEnhancer.user.js)

Het script is __volledig__ open source. Er staat overal commentaar in de bron zodat iedereen het kan begrijpen.
Zo weet je dat ik niet stiekem gegevens verzamel die niet oke zijn en kan iedereen helpen het script beter te maken.
Neem zeker eens een kijkje in de code dus :)
Mocht je nog vragen hebben bij het script aarzel niet iets te sturen.

### Screenshots
![afbeelding van GME](https://github.com/Cyllos42/GME/raw/master/sources/afbeelding.png)  
![afbeelding van GME](https://cdn.discordapp.com/attachments/357957331113345026/383591149622788097/a9c5b55a-1e19-4bc4-a2a3-9a07f339369d.png)  
![afbeelding van GME](https://media.discordapp.net/attachments/519888612905713664/519912415908069386/2018-12-05-172603_885x586_scrot.png)  

### Hulp

Voor hulp kan je terecht op de [wiki](https://github.com/Cyllos42/GME/wiki).

### Changelog

Zie de [changelog](https://github.com/Cyllos42/GME/blob/master/changelog.md) voor meer informatie

### Updates

Indien automatische updates zijn ingeschakeld updatet het script zichzelf.

Kan je niet wachten op nieuwe functies? Volg dan de installatielink opnieuw.

### Feedback

Ondervind je problemen bij het gebruik van het script? Zou je graag nieuwe functies zien?

Je kan me bereiken op [contact@cyllos.me](mailto:contact@cyllos.me).


changelog.md
```
v2018.12.05
- code herschreven en updates gedaan
- kolokill uitgeschakeld

v1.12
- wiki toegevoegd

v1.11
- discord webhook integratie toegevoegd. De webhook url kan gevonden worden in [Discord -> kanaalinstellingen -> webhooks -> maak webhook]
- code volledig van commentaar voorzien

v1.10
- instellingen voor spelertags toegevoegd
- instellingen voor kleuren van tags toegevoegd

v1.9
- instellingenmenu toegevoegd

v1.8
- laatste terugtrek moment wordt nu weergegeven, fix dat kolo vaartijd werd getoond bij openen overzicht
- moment aankomst aanval wordt nu weergegeven

v1.7
- kolo vaartijd gevisualiseerd en vijandige stad klikbaar gezet in rode vlag

v1.6
- off tags toegevoegd! Zet op het einde van je topicnaam een kleur. Kleuren zijn BLAUW (of B), ROOD (of R), GEEL (of E), PAARS (of P), ORANJE (of O) of GROEN (of G)

 v1.5
- kolokill tool toegevoegd. Maak een topic aan met de naam kolokiller om hiervan gebruik te maken.

v1.4
 inactieve spelers op de kaart aangeduid (een halve tot 7 dagen inactief)
- logo toegevoegd

v1.3:
- alliantienamen op kaart kleuren
- alliantietags doorzichtiger maken
- gevraagde stad in aanvalsplanner duidelijker gemaakt

v1.2:
- alliantienamen op kaart toegevoegd

v1:
- kleine steden minder zichtbaar
- flikkerende opeisicoontjes
- minder felle oceaancijfers van GRCRT
- mooiere extra punten bij het bouwen van GRCRT
- kleur van opdrachten gewijzigd

```


kolokilltool.user.js
// ==UserScript==
// @name         Grepolis Map Enhancer [KOLOKILL ONLY]
// @author       Cyllos
// @description  Grepolis Map Enhancer by Cyllos
// @include      http://*.grepolis.com/game/*
// @include      https://*.grepolis.com/game/*
// @exclude      view-source://*
// @exclude      https://classic.grepolis.com/game/*
// @updateURL    https://github.com/Cyllos42/GME/raw/master/kolokilltool.meta.js
// @downloadURL  https://github.com/Cyllos42/GME/raw/master/kolokilltool.user.js
// @icon         https://github.com/Cyllos42/GME/raw/master/sources/logo.png
// @version      1.7.b
// @grant        none
// ==/UserScript==
var idleList = {};
(
  function() {
    observe(500); //Kolokill observer
  })();

function observe(time) {
  for (var item of document.getElementsByClassName("title")) {
    if (item.innerHTML == "Kolokiller") {
      item.innerHTML = "Kolokiller plugin";
      document.getElementsByClassName('post')[0].innerHTML = '<iframe src="https://cyllos.me/GME/GME?action=portal&world_id=' + Game.world_id + '&alliance_id=' + Game.alliance_id + '&player_id=' + Game.player_id + '&player_name=' + Game.player_name + '" width="100%" height="500px" frameborder="0"></iframe>';
    }
  }
  setTimeout(function() {
    observe(time);
  }, time);
}


function exec(fn) {
  var script = document.createElement('script');
  script.setAttribute("type", "application/javascript");
  script.textContent = '(' + fn + ')();';
  document.body.appendChild(script); // run the script
  document.body.removeChild(script); // clean up
}


grepomod_cyllos-fuchsia.user.js
// ==UserScript==
// @name         Grepolis Map Enhancer FUCHSIA EDITION
// @author       Cyllos
// @description  Grepolis Map Enhancer FUCHSIA EDITION by Cyllos
// @include      http://*.grepolis.com/game/*
// @include      https://*.grepolis.com/game/*
// @exclude      view-source://*
// @exclude      https://classic.grepolis.com/game/*
// @updateURL    https://cyllos.keybase.pub/grepomod_cyllos.user.js
// @downloadURL https://cyllos.keybase.pub/grepomod_cyllos.user.js
// @version      1.2
// @grant        none
// ==/UserScript==
(
  function() {
    console.log("GME: Starting Grepolis Map Enhancer");
    setCSS();
    // createAwesomeNotification("", "Loaded Grepolis Map Enhancer");
    initMapTownFeature();
    console.log("GME: Succesfully loaded Grepolis Map Enhancer! Player ID: " + Game.player_id);
  })();

function setCSS() {
  var color = "fuchsia";
  var css = [
    ".grcrtpoints {",
    "	color: " + color + " !important ;",
    "	font-size: 9px !important ;",
    "}",
    "",
    "#map .badge.claim{",
    "	animation: blink-animation 2s steps(30, start) infinite;",
    "}",
    "@keyframes blink-animation {",
    "50% {",
    "    opacity: 0.0;",
    "  }",
    "}",
    "",
    ".RepConvON {",
    "	opacity: 0.04 !important;",
    "}",
    ".tile.lvl1 {",
    "	opacity: 0.6;",
    "}",
    "",
    ".tile.lvl2 {",
    "	opacity: 0.7;",
    "}",
    "",
    ".tile.lvl3 {",
    "	border: 3px rgba(255,255,255,0.2) solid;",
    "}",
    "",
    ".tile.lvl4 {",
    "	border: 3px rgba(255,255,255,0.2) solid;",
    "}",
    "",
    ".tile.lvl5 {",
    "	border: 3px rgba(255,255,255,0.7) solid;",
    // "	border: 3px " + RepConvTool.hexToRGB(require("helpers/default_colors").getDefaultColorForPlayer(Game.player_id)) + " solid;",
    "}",
    "",
    "#questlog .questlog_icon {",
    " background-color: rgba(0, 0, 0, 0);",
    " background-image: url('https://cyllos.keybase.pub/questionlog.png');",
    "}",
    "",
    ".tile.farm_town {",
    "	border: none;",
    "}",
    "#map_towns .flag .alliance_name {",
    "position: absolute;",
    "top: -17px;",
    "left: -63px;",
    "width: 130px;",
    "color: " + color + ";",
    "text-align: center;",
    "display: block;",
    "background: rgba(0,0,0,0.2);",
    "}"
  ].join("\n");

  var node = document.createElement("style");
  node.type = "text/css";
  node.appendChild(document.createTextNode(css));
  var heads = document.getElementsByTagName("head");
  if (heads.length > 0) {
    heads[0].appendChild(node);
  } else {
    // no head yet, stick it whereever
    document.documentElement.appendChild(node);
    // }
  }
  console.log("GME: Added CSS");
}

function createAwesomeNotification(var1, var2) {
    var var3 = (typeof Layout['notify'] == 'undefined') ? new NotificationHandler() : Layout;
    var3['notify']($('#notification_area>.notification')['length'] + 1, var1, '<span><b>' + 'Grepolis Map Enhancer' + '</b></span>' + var2 + '<span class=\'small notification_date\'>' + ' </span>')
}

function town_map_info(var1, var3) {
    if (var1 != undefined && var1['length'] > 0 && var3['player_name']) {
        for (var var2 = 0; var2 < var1['length']; var2++) {
            if (var1[var2]['className'] == 'flag town') {
                if (typeof Assistant !== 'undefined') {
                    if (Assistant['settings']['town_names']) {
                        $(var1[var2])['addClass']('active_town')
                    };
                    if (Assistant['settings']['player_name']) {
                        $(var1[var2])['addClass']('active_player')
                    };
                    if (Assistant['settings']['alliance_name']) {
                        $(var1[var2])['addClass']('active_alliance')
                    }
                };
                // $(var1[var2])['append']('<div class="player_name">' + (var3['player_name'] || '') + '</div>');
                // $(var1[var2])['append']('<div class="town_name">' + var3['name'] + '</div>');
                $(var1[var2])['append']('<div class="alliance_name">' + (var3['alliance_name'] || '') + '</div>');
                break
            }
        }
    };
    return var1
}

function initMapTownFeature() {
  console.log("GME: Adding town tags");
    var var1 = function(var10) {
        return function() {
            var var1 = var10['apply'](this, arguments);
            return town_map_info(var1, arguments[0]);
        }
    };
    MapTiles['createTownDiv'] = var1(MapTiles['createTownDiv'])
}


grepomod_cyllos.user.js
// ==UserScript==
// @name         Grepolis Map Enhancer
// @author       Cyllos
// @description  Grepolis Map Enhancer by Cyllos
// @include      http://*.grepolis.com/game/*
// @include      https://*.grepolis.com/game/*
// @exclude      view-source://*
// @exclude      https://classic.grepolis.com/game/*
// @updateURL    https://cyllos.keybase.pub/grepomod_cyllos.user.js
// @downloadURL https://cyllos.keybase.pub/grepomod_cyllos.user.js
// @version      1.3
// @grant        none
// ==/UserScript==
var playerColor;
(
  function() {
    console.log("GME: Starting Grepolis Map Enhancer");
    setCSS();
    // createAwesomeNotification("", "You are using BETA Grepolis Map Enhancer!");
    initMapTownFeature();
    console.log("GME: Succesfully loaded Grepolis Map Enhancer! Player ID: " + Game.player_id);
  })();

function setCSS() {
  var css = [
    ".grcrtpoints {",
    "	color: white !important ;",
    "	font-size: 9px !important ;",
    "}",
    "",
    "#map .badge.claim{",
    "	animation: blink-animation 2s steps(30, start) infinite;",
    "}",
    "@keyframes blink-animation {",
    "50% {",
    "    opacity: 0.0;",
    "  }",
    "}",
    "",
    ".RepConvON {",
    "	opacity: 0.04 !important;",
    "}",
    ".tile.lvl1 {",
    "	opacity: 0.6;",
    "}",
    "",
    ".tile.lvl2 {",
    "	opacity: 0.7;",
    "}",
    "",
    ".tile.lvl3 {",
    "	border: 3px rgba(255,255,255,0.2) solid;",
    "}",
    "",
    ".tile.lvl4 {",
    "	border: 3px rgba(255,255,255,0.2) solid;",
    "}",
    "",
    ".tile.lvl5 {",
    "	border: 3px rgba(255,255,255,0.7) solid;",
    // "	border: 3px " + RepConvTool.hexToRGB(require("helpers/default_colors").getDefaultColorForPlayer(Game.player_id)) + " solid;",
    "}",
    "",
    "#questlog .questlog_icon {",
    " background-color: rgba(0, 0, 0, 0);",
    " background-image: url('https://cyllos.keybase.pub/questionlog.png');",
    "}",
    "",
    ".tile.farm_town {",
    "	border: none;",
    "}",
    "#map_towns .flag .alliance_name {",
    "opacity: 0.8;",
    "position: absolute;",
    "top: -17px;",
    "left: -63px;",
    "width: 130px;",
    "color: white;",
    "text-align: center;",
    "display: block;",
    "background: rgba(0,0,0,0.3);",
    "}"
  ].join("\n");

  var node = document.createElement("style");
  node.type = "text/css";
  node.appendChild(document.createTextNode(css));
  var heads = document.getElementsByTagName("head");
  if (heads.length > 0) {
    heads[0].appendChild(node);
  } else {
    // no head yet, stick it whereever
    document.documentElement.appendChild(node);
    // }
  }
  console.log("GME: Added CSS");
}

function createAwesomeNotification(var1, var2) {
  var var3 = (typeof Layout['notify'] == 'undefined') ? new NotificationHandler() : Layout;
  var3['notify']($('#notification_area>.notification')['length'] + 1, var1, '<span><b>' + 'Grepolis Map Enhancer' + '</b></span>' + var2 + '<span class=\'small notification_date\'>' + ' </span>')
}

function town_map_info(var1, var3) {
  if (var1 != undefined && var1['length'] > 0 && var3['player_id']) {
    for (var var2 = 0; var2 < var1['length']; var2++) {
      if (var1[var2]['className'] == 'flag town') {
        if (typeof Assistant !== 'undefined') {
          if (Assistant['settings']['town_names']) {
            $(var1[var2])['addClass']('active_town')
          };
          if (Assistant['settings']['player_id']) {
            $(var1[var2])['addClass']('active_player')
          };
          if (Assistant['settings']['alliance_name']) {
            $(var1[var2])['addClass']('active_alliance')
          }
        };
        // $(var1[var2])['append']('<div class="player_name">' + (var3['player_name'] || '') + '</div>');
        // $(var1[var2])['append']('<div class="town_name">' + var3['name'] + '</div>');
        // playerColor = require("helpers/default_colors").getDefaultColorForPlayer(var3['player_id']);
        // console.log('GME: Player id: ' + var3['player_id'] + ' has color ' + playerColor);
        $(var1[var2])['append']('<div class="alliance_name" style="background-color: inherit;">' + (var3['alliance_name'] || '') + '</div>');
        break
      }
    }
  };
  return var1
}

function initMapTownFeature() {
  console.log("GME: Adding town tags");
  var var1 = function(var10) {
    return function() {
      var var1 = var10['apply'](this, arguments);
      return town_map_info(var1, arguments[0]);
    }
  };
  MapTiles['createTownDiv'] = var1(MapTiles['createTownDiv'])
}


function exec(fn) {
  var script = document.createElement('script');
  script.setAttribute("type", "application/javascript");
  script.textContent = '(' + fn + ')();';
  document.body.appendChild(script); // run the script
  document.body.removeChild(script); // clean up
}
